### DISCUSSION TITLE
Closed

### DISCUSSION THREAD

**Discussion by Vishwas-verma** [adonisjs/core#2865] [Help]
**Situtuation-**
 I want to list total_rating of all drivers where total_rating is average of all rating.

Driver **hasMany** bookings , booking **hasOne** Review(id,rating,timestamp) .

```
    return Driver.query().preload("bookings", b => b.withAggregate("review", r => {
      r.avg("rating").as("total_rating");
    }));
```

```
Driver.query().innerJoin("bookings",(b)=>{
      b.innerjoin("review", (c) => {
             c.on("some_condition")
       }).on("bookings.id", "=", "driver.booking_id");
    }).;
```
Why we cannot query like this as in laravel.
I know it will avg  booking ratings but how to achieve desired result ? Also how to write nested Join in this ?

**Comment by Vishwas-verma** [comment]
Found please see this discussion.
https://github.com/adonisjs/core/discussions/2874

**Reply by thetutlage** [reply]
Not sure why you have to created another discussion

**Reply by Vishwas-verma** [reply]
Hey, @thetutlage the other discussion contains the main issue, the need of this discussion is to get the approach ideas .
P.S please have a look at discussion you will
have clear idea about it.
